drivers/fs.go

```diff
@@ -0,0 +1,228 @@
+package drivers
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"time"
+)
+
+type FSOS struct {
+	baseURI  *url.URL
+	sessions map[string]*FSSession
+	lock     sync.RWMutex
+}
+
+type FSSession struct {
+	os     *FSOS
+	path   string
+	ended  bool
+	dCache map[string]*dataCache
+	dLock  sync.RWMutex
+}
+
+func NewFSDriver(baseURI *url.URL) *FSOS {
+	return &FSOS{
+		baseURI:  baseURI,
+		sessions: make(map[string]*FSSession),
+		lock:     sync.RWMutex{},
+	}
+}
+
+func (ostore *FSOS) NewSession(path string) OSSession {
+	ostore.lock.Lock()
+	defer ostore.lock.Unlock()
+	if session, ok := ostore.sessions[path]; ok {
+		return session
+	}
+	session := &FSSession{
+		os:     ostore,
+		path:   path,
+		dCache: make(map[string]*dataCache),
+		dLock:  sync.RWMutex{},
+	}
+	ostore.sessions[path] = session
+	return session
+}
+
+func (ostore *FSOS) GetSession(path string) *FSSession {
+	ostore.lock.Lock()
+	defer ostore.lock.Unlock()
+	if session, ok := ostore.sessions[path]; ok {
+		return session
+	}
+	return nil
+}
+
+func (ostore *FSOS) UriSchemes() []string {
+	return []string{""}
```

Comment:
Could have a scheme like `local` as well. Or is the idea that something like `/path/to/folder` can already be parsed as a scheme-less URL? 😮 (nice)

Reply:
Yes, the idea is to support normal file system paths.